vcd/resource_vcd_vapp_vm.go

```diff
@@ -29,8 +25,43 @@ const (
 	vappVmType       typeOfVm = "vcd_vapp_vm"
 )
 
-func resourceVcdVAppVm() *schema.Resource {
+// Maintenance guide for VM code
+//
+// VM codebase grew to be quite complicated because of a few reasons:
+// * It is a dated resource with many API quirks
```

Comment:
Is this "dated" statement still correct after this refactoring?

p.s. Great big comment block-guide overall!

Reply:
True. This `dated` doesn't give any real value in this context.